packages/wallets/src/phantom/phantom.page.ts

```diff
@@ -0,0 +1,121 @@
+      this.page = await this.browserContext.newPage();
+      await this.page.goto(this.extensionUrl + '/onboarding.html');
+      if (!this.page) throw "Page isn't ready";
+      const firstTime =
+        (await this.page
+          .locator('button:has-text("I already have a wallet")')
+          .count()) > 0;
+      if (firstTime) await this.firstTimeSetup();
+    });
+  }
+
+  async unlock() {
+    await test.step('Unlock', async () => {
+      if (!this.page) throw "Page isn't ready";
+      if ((await this.page.locator('id=password').count()) > 0) {
+        await this.page.fill('input[name=password]', this.config.PASSWORD);
+        await this.page.click('text=Unlock');
+      }
+    });
+  }
+
+  async closePopover() {
+    await test.step('Close popover if exists', async () => {
+      if (!this.page) throw "Page isn't ready";
+      const popover =
+        (await this.page.locator('data-testid=popover-close').count()) > 0;
+      if (popover) await this.page.click('data-testid=popover-close');
+    });
+  }
+
+  async firstTimeSetup() {
+    await test.step('First time setup', async () => {
+      if (!this.page) throw "Page isn't ready";
+      await this.page.click('button:has-text("I already have a wallet")');
+      const inputs = this.page.locator('.sc-bdvvtL >> input');
+      const seedWords = this.config.SECRET_PHRASE.split(' ');
+      for (let i = 0; i < seedWords.length; i++) {
+        await inputs.nth(i).fill(seedWords[i]);
+      }
+      await this.page.click('button[type=submit]');
+      await this.page.click('button:has-text("Import Selected Accounts")');
+      await this.page.fill('input[name=password]', this.config.PASSWORD);
+      await this.page.fill('input[name=confirmPassword]', this.config.PASSWORD);
+      await this.page.click('input[type=checkbox]');
+      await this.page.click('button:has-text("Continue")');
+      await this.page.click('button:has-text("Continue")');
+      await this.closePopover();
+    });
+  }
+
+  async importKey(key: string) {
+    await test.step('Import key', async () => {
+      if (!this.page) throw "Page isn't ready";
+      await this.navigate();
+      await this.page.click('.sc-bYoBSM bBxQWq');
+      await this.page.click('div:has-text("Wallet 1")');
+      await this.page.click('button:has-text("Add / Connect Wallet")');
+      await this.page.click(
+        'div[data-testid="add-account-create-new-wallet-button"]',
+      );
+      await this.page.fill('textarea', key);
+      await this.page.click('button:has-text("Import")');
+    });
+  }
+
+  async connectWallet(page: Page) {
+    await test.step('Connect wallet', async () => {
+      await page.click('button:has-text("Connect")');
+    });
+  }
+
+  async assertTxAmount(page: Page, expectedAmount: string) {
+    await test.step('Assert TX Amount', async () => {
+      const sendAmount = String(
+        await page.textContent('.sc-hKwDye byucnr'),
```

Comment:
another dynamic locator

Reply:
The same,cannot get rid of this
![image](https://user-images.githubusercontent.com/19698566/205675269-c92da7ce-713b-4bd7-9fbf-f5b1dc635eae.png)

Reply:
You can try to find this element with the expected text and SOL amount
If no element, the assertion should fail

Reply:
Fixed